moveit_ros/perception/pointcloud_octomap_updater/include/moveit/pointcloud_octomap_updater/pointcloud_octomap_updater.h

```diff
@@ -54,30 +54,31 @@ class PointCloudOctomapUpdater : public OccupancyMapUpdater
   PointCloudOctomapUpdater();
   ~PointCloudOctomapUpdater() override;
 
-  bool setParams(XmlRpc::XmlRpcValue& params) override;
+  bool setParams(const std::string& name_space) override;
 
-  bool initialize() override;
+  bool initialize(const rclcpp::Node::SharedPtr node) override;
   void start() override;
   void stop() override;
   ShapeHandle excludeShape(const shapes::ShapeConstPtr& shape) override;
   void forgetShape(ShapeHandle handle) override;
 
 protected:
-  virtual void updateMask(const sensor_msgs::PointCloud2& cloud, const Eigen::Vector3d& sensor_origin,
+  virtual void updateMask(const sensor_msgs::msg::PointCloud2& cloud, const Eigen::Vector3d& sensor_origin,
                           std::vector<int>& mask);
 
 private:
   bool getShapeTransform(ShapeHandle h, Eigen::Isometry3d& transform) const;
-  void cloudMsgCallback(const sensor_msgs::PointCloud2::ConstPtr& cloud_msg);
+  void cloudMsgCallback(const sensor_msgs::msg::PointCloud2::ConstSharedPtr& cloud_msg);
   void stopHelper();
 
-  ros::NodeHandle root_nh_;
-  ros::NodeHandle private_nh_;
+  // ros::NodeHandle root_nh_;
```

Comment:
Do we need a separate private node? If yes, please add a TODO, otherwise these comments should be removed

Reply:
I'm not quite sure whether we need this private node or not. The private node is used to create a publisher that publishes the point cloud filtering out the robot arm. The publish happens within the callback function of a raw point cloud topic subscriber created from the root node.